libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx

```diff
@@ -155,97 +155,100 @@ function GrantAccess() {
-                </span>
-              ) : name ? (
-                <button
-                  disabled={loading}
-                  onClick={clearForm}
-                  className={styles.icon}
-                >
-                  <Icon icon="close" size="large" color="blue400" />
-                </button>
-              ) : null}
-            </div>
-          </GridColumn>
-          <GridColumn span={['12/12', '12/12', '4/12']}>
-            <Button
-              size="large"
-              fluid
-              type="submit"
-              icon="arrowForward"
-              disabled={!name || loading}
-              loading={loading}
-            >
-              {formatMessage({
-                id: 'service.portal.settings.accessControl:grant-form-submit',
-                defaultMessage: 'Áfram',
-              })}
-            </Button>
-          </GridColumn>
+                        validate: {
+                          value: (value: number) => {
+                            if (
+                              value.toString().length === 10 &&
+                              !kennitala.isValid(value)
+                            ) {
+                              return formatMessage({
+                                id:
+                                  'service.portal.settings.accessControl:grant-invalid-ssn',
+                                defaultMessage:
+                                  'Kennitalan er ekki gild kennitala',
+                              })
+                            }
+                          },
+                        },
+                      } as ValidationRules
+                    }
+                    type="tel"
+                    format="######-####"
+                    label={formatMessage(sharedMessages.nationalId)}
+                    placeholder={formatMessage(sharedMessages.nationalId)}
+                    error={errors.toNationalId?.message}
+                    onChange={(value) => {
+                      requestDelegation(value)
+                    }}
+                    size="xs"
+                  />
+                </Box>
+                {queryLoading ? (
+                  <span
+                    className={cn(styles.icon, styles.loadingIcon)}
+                    aria-label="Loading"
+                  >
+                    <Icon icon="reload" size="large" color="blue400" />
+                  </span>
+                ) : name ? (
+                  <button
+                    disabled={loading}
+                    onClick={clearForm}
+                    className={styles.icon}
+                  >
+                    <Icon icon="close" size="large" color="blue400" />
+                  </button>
+                ) : null}
+              </div>
+            </GridColumn>
+            <GridColumn span={['12/12', '12/12', '4/12']}>
+              <Button
+                size={name ? 'small' : 'large'}
```

Comment:
?

Reply:
It's terrible - don't look (but I can screen record if you like). I'm waiting on "NumberFormat" package to release a fix so that I can change the size variant for numbered inputs. This beautiful hack is only temporary 😬 

You can see the problem here: https://github.com/s-yadav/react-number-format/pull/577

Reply:
Haha, you're right. It's terrible! 😅

Do we want to get rid of all of the large inputs in the service portal? I thought it was only in the places where we wanted to save space or take less attention. I would argue that this is the primary thing on this screen so it can use the "standard" large input.

Reply:
Also, there's a layout regression when showing a validation error:
![image](https://user-images.githubusercontent.com/115094/146940086-ca8670b3-b03f-4e20-acb8-af17e6391d0e.png)

Reply:
Yes, that is the goal. Although I kinda agree when theres only one input. Maybe it can stay - that would solve this for now. Another thing to talk to Keli about.

Reply:
Done ✅